docs/performances.md

```diff
@@ -0,0 +1,169 @@
+  - this can be any string. Visualizing tools tends create as many category as provided (hence their are some exceptions like ReactNative's flipper plugin).
+
+Example of measuring the elapsed time between the call of CozyClient's `query()` method and its ending or its failing block
+```js
+async query(queryDefinition, { update, executeFromStore, ...options } = {}) {
+  const markName = client.mark(`client.query(${queryDefinition.doctype})`)
+
+  try {
+    // ... method implementation
+
+    this.performanceApi.measure({
+      markName: markQuery,
+      measureName: `${markName} success`,
+      category: 'CozyClient queries'
+    })
+  } catch (error) {
+    this.performanceApi.measure({
+      markName: markQuery,
+      measureName: `${markName} error`,
+      category: 'CozyClient queries',
+      color: 'error'
+    })
+  }
+}
+```
+
+## Adding measurements
+
+On initial implementation, we chose to measure some parts of the code that we estimated useful and with potential high impact on performances (i.e. queries, store management, pouch management etc.)
+
+We excpect to need more measurements in the future but we need to care to keep the balance beetween having useful measurements and to not visualy overload the measurement graphs
+
+The API is meant to be simple to use so measurements can be quickly added temporarily in development environements
+
+In the scenario where a measurement seems to be useful enough to be permanently added to our code base, please consider discussing this choice with the development team.
+
+## Web usage
+
+On web environement, it is possible to inject the already provided `webPerformanceAPI` implementation of the API
+
+Example of CozyClient initialization with web compatible version of the PerformanceAPI
+```js
+import CozyClient, { webPerformanceAPI } from 'cozy-client'
+
+const client = new CozyClient({
+  uri,
+  schema,
+  token,
+  performancesApi: webPerformanceAPI
+})
+```
+
+Then CozyClient will use this implementation to do its internal measurements
+
+The consuming app can also add its own measurements by either:
+- Calling `client.performanceAPI.mark()` and `client.performanceAPI.measure()` method
+- Directly calling the web native API `performance.mark()` and `performance.measure()`
+  - However, please keep in mind that if both APIs will produce the same output, their interface is slightly different. So for code homogeneity purpose, please prefer using the CozyClient's API
+
+When the PerformanceAPI is used, then the result can be visualized in the browser's Performance develools
+
+![Performance devtools screenshot](./performances_devtools.png)
+
+## Other platforms
+
+No implementation is provided by CozyClient for other platforms. So each consuming app needs to provide their own implementation of the API
+
+The API interface is described in the JSDoc
+
+Example of custom implementations:
+- Flagship App: [](TODO ADD LINK)
```

Comment:
Will we think of this? :smile:

Reply:
For sure, the task is on the PR description's todo list, so I should not forget to do it (I won't merge this without adding the link).

Reply:
Done: https://github.com/cozy/cozy-client/compare/fabe643260d358b4cd558200d45dec15387e476b..2d7973e259ca4c6eb6e1c134e0dad87875518288